### PR TITLE
Sync code

### PR DESCRIPTION
**Pull request by bxq2011hust**

### CODE REVIEW

File: libstorage/MemoryTable2.cpp [hunk 1]
```diff
@@ -214,7 +225,8 @@ int MemoryTable2::insert(
     }
     catch (std::exception& e)
     {
-        STORAGE_LOG(ERROR) << LOG_BADGE("MemoryTable2")
+        // impossible, so exit
+        STORAGE_LOG(FATAL) << LOG_BADGE("MemoryTable2")
```
Comment (wheatli) [inline libstorage/MemoryTable2.cpp]: why not use prepare_exit here ?

File: libledger/DBInitializer.cpp [hunk 2]
```diff
@@ -173,11 +173,9 @@ void DBInitializer::initSQLStorage()
     sqlStorage->setChannelRPCServer(m_channelRPCServer);
     sqlStorage->setTopic(m_param->mutableStorageParam().topic);
     sqlStorage->setFatalHandler([](std::exception& e) {
-        (void)e;
-        LOG(FATAL) << "Access amdb failed, exit";
-
-        // must use exit to avoid rasie failed
-        exit(1);
+        LOG(ERROR) << "Access amdb failed exit:" << e.what();
```
Comment (wheatli) [inline libledger/DBInitializer.cpp]: why not use FATAL

File: libstorage/MemoryTable2.cpp (hunk 1 above)
Comment (bxq2011hust) [reply libstorage/MemoryTable2.cpp]: This is pure memory operation, should never failed, if then must be  core

File: libledger/DBInitializer.cpp (hunk 2 above)
Comment (bxq2011hust) [reply libledger/DBInitializer.cpp]: This error is possible, our code should handle this, make process exit normally with error log.

File: libconsensus/pbft/PBFTEngine.cpp [hunk 3]
```diff
@@ -241,16 +241,16 @@ void PBFTEngine::backupMsg(std::string const& _key, PBFTMsg const& _msg)
     }
     catch (DatabaseError const& e)
     {
-        PBFTENGINE_LOG(FATAL) << LOG_BADGE("DatabaseError")
+        PBFTENGINE_LOG(ERROR) << LOG_BADGE("DatabaseError")
```
[the remark this answers is not in the document]
Comment (bxq2011hust) [reply libconsensus/pbft/PBFTEngine.cpp]: This error is possible, our code should handle this, make process exit normally with error log.

**Pull request merged**